### PR TITLE
 SWARM-945 | OpenTracing Hawkular jaxrs fraction

### PR DESCRIPTION
**Pull request by pavolloffay**
https://issues.jboss.org/browse/HWKAPM-779

Please do not review yet!

### CODE REVIEW

**Comment by pavolloffay** [comment]
This should wait until https://github.com/opentracing-contrib/java-jaxrs/pull/2 and https://issues.jboss.org/browse/HWKAPM-793 are merged and new versions are released.

**Comment by pavolloffay** [comment]
@kenfinnigan could you please review? 

Example app is here https://github.com/wildfly-swarm/wildfly-swarm-examples/pull/120

File: pom.xml [hunk 1]
```diff
@@ -1222,6 +1234,7 @@
     <module>fractions/microprofile</module>
     <module>fractions/monitor</module>
     <module>fractions/netflix</module>
+    <module>fractions/opentracing-hawkular-jaxrs</module>
```
Comment (kenfinnigan) [inline pom.xml]: This module needs to be in the "unsupported" profile

**Comment by kenfinnigan** [comment]
ok to test

**Comment by gastaldi** [comment]
Retest this please

File: fractions/opentracing-hawkular-jaxrs/pom.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      <groupId>org.wildfly.swarm</groupId>
+      <artifactId>spi</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.jboss</groupId>
+      <artifactId>staxmapper</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
```
Comment (kenfinnigan) [inline fractions/opentracing-hawkular-jaxrs/pom.xml:53]: Where is jboss-dmr dep needed?

File: fractions/opentracing-hawkular-jaxrs/pom.xml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      <artifactId>javax.inject</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>javax.enterprise</groupId>
+      <artifactId>cdi-api</artifactId>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
```
Comment (kenfinnigan) [inline fractions/opentracing-hawkular-jaxrs/pom.xml:64]: The fraction depends on CDI?

Comment (pavolloffay) [reply fractions/opentracing-hawkular-jaxrs/pom.xml:64]: yes

File: fractions/opentracing-hawkular-jaxrs/pom.xml (hunk 2 above)
Comment (pavolloffay) [reply fractions/opentracing-hawkular-jaxrs/pom.xml:53]: It works without it. I followed the same approach as in zipkin fraction.

**Comment by pavolloffay** [comment]
@kenfinnigan PR updated, dependency on jboss-drm has been removed. About dependencies, this follows the same code from zipkin fraction by @heiko-braun

**Comment by kenfinnigan** [comment]
zipkin fraction does not have a dependency on org.wildfly.swarm:cdi, but this does.

I'm trying to find out whether it's mandatory that user applications use CDI for this fraction to work

**Comment by pavolloffay** [comment]
I tried to remove swarm:cdi. I'm getting this https://paste.fedoraproject.org/559071/14871876/ 
```
...
[INFO]  - requires: javax.ws.rs.api:main
[INFO]  - requires: org.jboss.as.jaxrs:main
[INFO]  - requires: org.jboss.resteasy.resteasy-jaxrs:main
[ERROR] Unable to locate required module: org.wildfly.swarm.cdi:main
java.io.IOException: Unable to locate required module: org.wildfly.swarm.cdi:main
	at org.wildfly.swarm.plugin.process.ModuleFiller.locateFillModules(ModuleFiller.java:157)
	at org.wildfly.swarm.plugin.process.ModuleFiller.apply(ModuleFiller.java:108)
	at org.wildfly.swarm.plugin.process.ProcessMojo.execute(ProcessMojo.java:59)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
...
``

Do you know where the problem could be?

**Comment by kenfinnigan** [comment]
it also needs to be removed from https://github.com/wildfly-swarm/wildfly-swarm/pull/250/files#diff-70a3e0a36bab7fc7f9b58561bbe3fe82R21

**Comment by pavolloffay** [comment]
excellent it works now :+1: updated and rebased

**Comment by pavolloffay** [comment]
Rebased again

**Review by kenfinnigan — APPROVED** [review]

**Comment by pavolloffay** [comment]
@kenfinnigan @heiko-braun thanks!

**Pull request merged**